docs/DEVELOPMENT.md

```diff
@@ -1,15 +1,169 @@
+
+1. Create a draft release on github, e.g. `vX.Y.Z`.
+2. Change `version` field in `package.json` to `X.Y.Z`.
+3. Update, commit and push `static/chat.delta.desktopp.appdata.xml`
+   with the new release information.
+4. Regenerate `package-lock.json` using `npm install`, commit and push
+   modified `package.json` and `package-lock.json` (repeat until
+   release is ready).
+5. Once done, publish the release on github, which will create the
+   tag.
+6. File an issue at
+   <https://github.com/flathub/chat.delta.desktop/issues> to make a new
+   release.
+
+Also see <https://www.electron.build/configuration/publish>
+
+### Code Structure <a id="code-structure"></a>
+
+Some important folders and files:
+
+```powershell
+├── _locales/                 # translation files in xml and json
+│   ├── _untranslated_en.json # can contain experimental language strings
+│   └── languages.json        # central file which keeps the human readable language
+├── .gihub/workflows          # source of our github actions
+├── bin/                      # various helper scripts
+│   └── build/                # Build scripts
+├── build/                    # files needed for electron-builder
+├── ci_scripts/               # scripts and dockerfiles used by the CI
+├── docs/                     # documentation
+├── images/                   # image files like icons or backgrounds
+├── index.js                  # entry point for the main process
+├── README_ASSETS/            # Images used in the readme file and documentation
+├── scss/                     # styelsheets which need preprocessing
+├── src/
+│   ├── main/                 # typescript for the main process
+│   ├── renderer/             # typescript for the renderer process
+│   └── shared/               # typescript that is shared between both processes
+├── static/
+│   ├── fonts/                # fonts
+│   ├── help/                 # the in-app help
+│   └── main.html             # main html file in renderer process
+├── test/
+│   ├── integration/          # integration tests
+│   ├── testcafe/             # testcafe tests
+│   └── unit/                 # unit tests
+└── themes/                   # default themes
+```
 
+## Tips for specific subjects <a id="specific-tipps"></a>
 
-### VS Code users
+### VS Code users <a id="vscode"></a>
 
 Problem: Strange Typscript errors that are only visible in vscode but the project compiles normaly
 
 Solution: Tell VS Code to use the workspace version of typescript intead af an buildt-in version [more info](https://code.visualstudio.com/docs/typescript/typescript-compiling#_why-do-i-get-different-errors-and-warnings-with-vs-code-than-when-i-compile-my-typescript-project)
 
-### URI Schemes on linux
+### URI Schemes on linux <a id="linux-uri-schemes"></a>
 
 can only be tested in builds that have a desktop file - simplest way is to install the appimage generated by `npx electron-builder --linux AppImage`. (Installing with AppImageLauncher)
```

Comment:
Isn't the appimage the one that currently doesn't support it because of a bug in electron-builder? Or is it fixed?

Reply:
this part here is not new, lets don't discuss it in this pr, but your right atm the appimage is broken in that regard.